lib/cocoapods/target/pod_target.rb

```diff
@@ -141,9 +141,12 @@ def product_module_name
     # A target should not be build if it has no source files.
     #
     def should_build?
-      source_files = file_accessors.flat_map(&:source_files)
-      source_files -= file_accessors.flat_map(&:headers)
-      !source_files.empty?
+      return @should_build if defined? @should_build
+      @should_build ||= begin
```

Comment:
the `||=` is redundant, it will never be used, this code will only ever be executed once per instance